### PR TITLE
Enabled the Hardware Isolation (aka Guard) feature for Core and Memory

### PR DESCRIPTION
**Pull request by RameshIyyar**
Created this PR to address the below comments.

- Enabled the Hardware Isolation (aka Guard) feature for Core and Memory #77 (comment)

**Note:** 
- This patch is required to enable the manual guard feature. 
- This PR is created on top of https://github.com/ibm-openbmc/bmcweb/pull/95 and https://github.com/ibm-openbmc/bmcweb/pull/96 so, please ignore the first two commits in this PR.

### CODE REVIEW

**Comment by RameshIyyar** [comment]
> * This PR is created on top of [Implement SubProcessors for processor core #95](https://github.com/ibm-openbmc/bmcweb/pull/95) and [Workaround to handle DCM (Dual-Chip Module) package for Redfish #96](https://github.com/ibm-openbmc/bmcweb/pull/96) so, please ignore the first two commits in this PR.

This PR should rebase once #96 PR is merged.

**Comment by RameshIyyar** [comment]
@gtmills please review

File: redfish-core/lib/processor.hpp [hunk 1]
```diff
@@ -1420,4 +1832,95 @@ inline void requestRoutesProcessor(App& app)
...
+inline void requestRoutesSubProcessorsCore(App& app)
+{
+    BMCWEB_ROUTE(
+        app, "/redfish/v1/Systems/system/Processors/<str>/SubProcessors/<str>")
+        .privileges(redfish::privileges::getProcessor)
+        .methods(boost::beast::http::verb::get)(
+            [](const crow::Request&,
+               const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+               const std::string& processorId, const std::string& coreId) {
+                getSubProcessorData(asyncResp, processorId, coreId);
```
Comment (gtmills) [inline redfish-core/lib/processor.hpp:1917]: We already did here: 
https://github.com/ibm-openbmc/bmcweb/pull/95/files
Can we rebase this commit?

File: redfish-core/include/utils/hw_isolation.hpp [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+            if (ec || objects.empty())
+            {
+                BMCWEB_LOG_ERROR << "DBus response error [" << ec.value()
+                                 << " : " << ec.message()
+                                 << "] when tried to check the given resource "
+                                    "is present in tne inventory";
+                messages::internalError(aResp->res);
+                return;
+            }
+
```
Comment (gtmills) [inline redfish-core/include/utils/hw_isolation.hpp:260]: Nitpick: Shouldn't objects.empty() be a resource not found? 
https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/44175/9/redfish-core/lib/systems.hpp#1989

File: redfish-core/include/utils/hw_isolation.hpp [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+            sdbusplus::message::object_path resourceObjPath;
+            for (auto& object : objects)
```
Comment (gtmills) [inline redfish-core/include/utils/hw_isolation.hpp:262]: const auto& object : objects ?

File: redfish-core/include/utils/hw_isolation.hpp (hunk 2 above)
Comment (RameshIyyar) [reply redfish-core/include/utils/hw_isolation.hpp:260]: Actually, I should remove this check (checking in LN272) - will remove `objects.empty()`

**Comment by RameshIyyar** [comment]
@gtmills please review, addressed your comments

File: redfish-core/lib/processor.hpp (hunk 1 above)
Comment (RameshIyyar) [reply redfish-core/lib/processor.hpp:1924]: Done, since #96 merged

File: redfish-core/include/utils/hw_isolation.hpp [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+                                             "NotAllowed") == 0)
+            {
+                messages::propertyNotWritable(aResp->res, "Enabled");
+            }
+            else if (strcmp(dbusError->name, "xyz.openbmc_project.Common.Error."
+                                             "Unavailable") == 0)
+            {
+                messages::resourceInStandby(aResp->res);
+            }
+            else if (strcmp(dbusError->name, "xyz.openbmc_project."
```
Comment (raviteja-b) [inline redfish-core/include/utils/hw_isolation.hpp:75]: std::strcmp?

File: redfish-core/lib/memory.hpp [hunk 5]
```diff
@@ -833,6 +834,53 @@ inline void getDimmPartitionData(std::shared_ptr<bmcweb::AsyncResp> aResp,
...
+ * @return The redfish response in the given buffer.
+ *
+ * @note - The "Enabled" member of the Memory (aka DIMM) is mapped with
+ *         "xyz.openbmc_project.Object.Enable::Enabled" dbus property.
+ */
+inline void getObjectEnable(std::shared_ptr<bmcweb::AsyncResp> aResp,
+                            const std::string& service, const std::string& path)
+{
+    crow::connections::systemBus->async_method_call(
+        [aResp{std::move(aResp)}](
```
Comment (raviteja-b) [inline redfish-core/lib/memory.hpp:854]: any reason for std::move here

File: redfish-core/lib/memory.hpp [hunk 6]
```diff
@@ -833,6 +834,53 @@ inline void getDimmPartitionData(std::shared_ptr<bmcweb::AsyncResp> aResp,
...
+ * @param[in] aResp - The redfish response to return.
+ * @param[in] service - The dbus service name which is hosting the given path.
+ * @param[in] path - The given Memory resource inventory dbus object path.
+ *
+ * @return The redfish response in the given buffer.
+ *
+ * @note - The "Enabled" member of the Memory (aka DIMM) is mapped with
+ *         "xyz.openbmc_project.Object.Enable::Enabled" dbus property.
+ */
+inline void getObjectEnable(std::shared_ptr<bmcweb::AsyncResp> aResp,
```
Comment (raviteja-b) [inline redfish-core/lib/memory.hpp:850]: use reference? & aRes

Comment (raviteja-b) [inline redfish-core/lib/memory.hpp:850]: use reference?

File: redfish-core/lib/memory.hpp [hunk 7]
```diff
@@ -833,6 +834,53 @@ inline void getDimmPartitionData(std::shared_ptr<bmcweb::AsyncResp> aResp,
...
+            {
+                if (propName == "Enabled")
+                {
+                    const bool* enabled = std::get_if<bool>(&propValue);
+                    if (enabled == nullptr)
+                    {
+                        messages::internalError(aResp->res);
+                        break;
+                    }
+                    aResp->res.jsonValue["Enabled"] = *enabled;
```
Comment (raviteja-b) [inline redfish-core/lib/memory.hpp:876]: break?

**Comment by rfrandse** [comment]
Common practice discussion for downstream PR's
ibm-openbmc/dev#3341

**Pull request merged**